### DISCUSSION TITLE
find bridgehead atoms?

### DISCUSSION THREAD

**Discussion by hstern2** [rdkit/rdkit#5052] [Q&A]
best/easiest way to identify bridgehead atoms?

**Comment by hstern2** [comment]
python translation of  `unsigned int calcNumBridgeheadAtoms` in Lipinski.cpp  ??


```py
from itertools import combinations

def _inc(a, k):
    if k not in a:
        a[k] = 0
    a[k] += 1

def BridgeheadAtoms(m):
    r = m.GetRingInfo()
    a = {}
    for i,j in combinations(r.BondRings(), 2):
        bridge = set(i) & set(j)
        if len(bridge) >= 2:
            for b in bridge:
                b = m.GetBondWithIdx(b)
                _inc(a, b.GetBeginAtomIdx())
                _inc(a, b.GetEndAtomIdx())
    return [i for i,n in a.items() if n == 1]
```

**Comment by DavidACosgrove** [comment]
I'd suggest using SMARTS.  I assume a bridgehead atom has 3 ring bonds off it, though maybe it could also be 4(?).  In which case:
```
mol = Chem.MolFromSmiles('C1C3CC2CC(CC1C2)C3')
qmol = Chem.MolFromSmarts('[x{3-4}]')
print(mol.GetSubstructMatches(qmol))
```
Gives
![image](https://user-images.githubusercontent.com/9198870/155973874-7aee790c-fb47-4fed-b5bf-7fe577ba8ac5.png)

Note that that uses the RDKit SMARTS extension allowing specification of a range.  In standard SMARTS you'd have to use 2 queries, [x3] and [x4].

**Reply by hstern2** [reply]
Thanks for the response.  This would need to be modified as it includes atoms with only one bond in the bridge e.g. naphthalene but yes maybe easier to do this with SMARTS

**Comment by DavidACosgrove** [comment]
This page http://www.chem.ucla.edu/~harding/IGOC/B/bridgehead.html says naphthalene does contain bridgehead bonds.  If you don't want aromatic hits, you could use [Ax3].

**Reply by greglandrum** [reply]
> This page http://www.chem.ucla.edu/~harding/IGOC/B/bridgehead.html says naphthalene does contain bridgehead bonds. If you don't want aromatic hits, you could use [Ax3].

The RDKit defintion intentionally avoids the napthalene case (which, of course, makes the definition more complicated and rules out a straightforward SMARTS)

**Comment by hstern2** [comment]
the terminology is confusing, I think.  rdkit seems to restrict "bridgehead atoms" to bridged, not fused, ring systems.  see https://www.rdkit.org/docs/source/rdkit.Chem.rdMolDescriptors.html
CalcNumBridgeheadAtoms

Returns the number of bridgehead atoms (atoms shared between rings that share at least two bonds)

 so for naphthalene (or decalin)

```
>>> from rdkit.Chem import MolFromSmiles
>>> from rdkit.Chem.rdMolDescriptors import CalcNumBridgeheadAtoms
>>> CalcNumBridgeheadAtoms(MolFromSmiles('c1c2ccccc2ccc1'))
0
```

Anyway, I'm currently using something like the code above, transcribed from Lipinski.cpp
